### ISSUE TITLE
Dreambooth example failing, probably due to dependency issue

### ISSUE THREAD

**Issue by LukeGahan** [modal-labs/modal-examples#402]
**tl;dr**: `dreambooth_app.py`  demo not running, would expect to run as is. Seems to be an issue with package versions but I can't figure out what the correct combination should be.

I'm trying to run the `dreambooth_app.py` (using: `modal run dreambooth_app.py`) but unfortunately I'm getting an error relating to the version of accelerate. I'm using the code exactly as is, I've just swapped in links to my own images.

```
Traceback (most recent call last):
  File "/pkg/modal/_container_entrypoint.py", line 351, in handle_input_exception
    yield
  File "/pkg/modal/_container_entrypoint.py", line 437, in run_inputs
    res = imp_fun.fun(*args, **kwargs)
  File "/root/dreambooth_app.py", line 206, in train
    from accelerate.utils import write_basic_config
  File "/usr/local/lib/python3.10/site-packages/accelerate/__init__.py", line 3, in <module>
    from .accelerator import Accelerator
  File "/usr/local/lib/python3.10/site-packages/accelerate/accelerator.py", line 33, in <module>
    from .checkpointing import load_accelerator_state, load_custom_state, save_accelerator_state, save_custom_state
  File "/usr/local/lib/python3.10/site-packages/accelerate/checkpointing.py", line 24, in <module>
    from .utils import (
  File "/usr/local/lib/python3.10/site-packages/accelerate/utils/__init__.py", line 119, in <module>
    from .megatron_lm import (
  File "/usr/local/lib/python3.10/site-packages/accelerate/utils/megatron_lm.py", line 32, in <module>
    from transformers.modeling_outputs import (
  File "/usr/local/lib/python3.10/site-packages/transformers/__init__.py", line 26, in <module>
    from . import dependency_versions_check
  File "/usr/local/lib/python3.10/site-packages/transformers/dependency_versions_check.py", line 57, in <module>
    require_version_core(deps[pkg])
  File "/usr/local/lib/python3.10/site-packages/transformers/utils/versions.py", line 117, in require_version_core
    return require_version(requirement, hint)
  File "/usr/local/lib/python3.10/site-packages/transformers/utils/versions.py", line 111, in require_version
    _compare_versions(op, got_ver, want_ver, requirement, pkg, hint)
  File "/usr/local/lib/python3.10/site-packages/transformers/utils/versions.py", line 44, in _compare_versions
    raise ImportError(
ImportError: accelerate>=0.20.3 is required for a normal functioning of this module, but found accelerate==0.19.0.
Try: pip install transformers -U or pip install -e '.[dev]' if you're working with git main
```

When I specify `accelerate>=0.20.3`  I get past the pip install part of the code but I get an error when subprocess is called:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/xformers/ops/fmha/triton.py", line 17, in <module>
    from flash_attn.flash_attn_triton import (
ModuleNotFoundError: No module named 'flash_attn

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/examples/dreambooth/train_dreambooth.py", line 32, in <module>
    import diffusers
  File "/root/src/diffusers/__init__.py", line 35, in <module>
    from .models import (
  File "/root/src/diffusers/models/__init__.py", line 19, in <module>
    from .autoencoder_kl import AutoencoderKL
  File "/root/src/diffusers/models/autoencoder_kl.py", line 23, in <module>
    from .vae import Decoder, DecoderOutput, DiagonalGaussianDistribution, Encoder
  File "/root/src/diffusers/models/vae.py", line 22, in <module>
    from .unet_2d_blocks import UNetMidBlock2D, get_down_block, get_up_block
  File "/root/src/diffusers/models/unet_2d_blocks.py", line 18, in <module>
    from .attention import AttentionBlock
  File "/root/src/diffusers/models/attention.py", line 22, in <module>
    from .cross_attention import CrossAttention
  File "/root/src/diffusers/models/cross_attention.py", line 25, in <module>
    import xformers.ops
  File "/usr/local/lib/python3.10/site-packages/xformers/ops/__init__.py", line 8, in <module>
    from .fmha import (
  File "/usr/local/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py", line 10, in <module>
    from . import cutlass, decoder, flash, small_k, triton
  File "/usr/local/lib/python3.10/site-packages/xformers/ops/fmha/triton.py", line 39, in <module>
    flash_attn = import_module_from_path(
  File "/usr/local/lib/python3.10/site-packages/xformers/ops/fmha/triton.py", line 36, in import_module_from_path
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1016, in get_code
  File "<frozen importlib._bootstrap_external>", line 1073, in get_data
FileNotFoundError: [Errno 2] No such file or directory: '/root/third_party/flash-attention/flash_attn/flash_attn_triton.py'
Traceback (most recent call last):
  File "/usr/local/bin/accelerate", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py", line 45, in main
    args.func(args)
  File "/usr/local/lib/python3.10/site-packages/accelerate/commands/launch.py", line 979, in launch_command
    simple_launcher(args)
  File "/usr/local/lib/python3.10/site-packages/accelerate/commands/launch.py", line 628, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
```

Installing flash_attn then I get an error about torch not being installed. So I then tried to install flash_attn after torch is installed and  then I get the following error:

```
     Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-ji7oryv0/flash-attn_4dc31d3a534f4bcbb9e25d023a0889bd/setup.py", line 108, in <module>
          raise_if_cuda_home_none("flash_attn")
        File "/tmp/pip-install-ji7oryv0/flash-attn_4dc31d3a534f4bcbb9e25d023a0889bd/setup.py", line 55, in raise_if_cuda_home_none
          raise RuntimeError(
      RuntimeError: flash_attn was requested, but nvcc was not found.  Are you sure your environment has nvcc available?  If you're installing within a container from https://hub.docker.com/r/pytorch/pytorch, only images whose names contain 'devel' will provide nvcc.

      Warning: Torch did not find available GPUs on this system.
       If your intention is to cross-compile, this is not an error.
      By default, Apex will cross-compile for Pascal (compute capabilities 6.0, 6.1, 6.2),
      Volta (compute capability 7.0), Turing (compute capability 7.5),
      and, if the CUDA version is >= 11.0, Ampere (compute capability 8.0).
      If you wish to cross-compile for a single specific architecture,
      export TORCH_CUDA_ARCH_LIST="compute capability" before running setup.py.`

```

Anyway, I'm sure the issue is due to some sort of mismatch in versions but I'm not able to figure out the correct combinaton. Does anyone happen to know what the correct setup is? Or is there some other underlying issue?

**Comment by egoodman92** [comment]
Found accelerate==0.19, transformers==4.20 to work (by sheer guessing). However even after that ran into:

EntryNotFoundError: 404 Client Error: Entry Not Found for url: 
https://huggingface.co/runwayml/stable-diffusion-v1-5/resolve/main/config.json

:( 

Trying to solve this now

**Comment by PradeeshSuganthan** [comment]
Still running into this issue but I managed to get it working, I used accelerate==0.19 and transformers==4.28 and had no issues after.

**Comment by yaneq** [comment]
I got it working by removing version specs accelerate. Then i got an error that dreambooth script in the diffusers repo returned a non-0 value. Fixed that by replacing

```
GIT_SHA = "ed616bd8a8740927770eebe017aedb6204c6105f"
```

with 

```
GIT_SHA = "01782c220eea24f74e149c0ab1153e74ba65711b"
```

Now it does not throw error anymore, but seems to run into a timeout. After ca. 3 minutes i get a local console error:
```
Runner terminated.
...
rate
launch
examples/dreambooth/train_dreambooth.py
--train_text_encoder
--pretrained_model_name_or_path=runwayml/stable-diffusion-v1-5
--instance_data_dir=/img
--output_dir=/model
--instance_prompt='a photo of Qwerty Golden Retriever'
--resolution=512
--train_batch_size=1
--gradient_accumulation_steps=1
--learning_rate=2e-06
--lr_scheduler=constant
--lr_warmup_steps=0
--max_train_steps=600
--checkpointing_steps=1000' returned non-zero exit status 1.
```

Tried to up the timeouts to 30 minutes, but that did not help yet.

**Comment by thundergolfer** [comment]
Merged https://github.com/modal-labs/modal-examples/pull/534 and now o0n latest `main` I was now able to get this running 
![image](https://github.com/modal-labs/modal-examples/assets/12058921/38004e9a-2535-4d4b-b020-1bb4386d8534)

I'll close this issue as fixed but please get back if there's still issues! 🙂
